jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/impl/PostgreSQLSqlDialect.java

```diff
@@ -298,6 +303,24 @@ public DataType dialectSpecificMapJdbcType(final JdbcTypeDescription jdbcTypeDes
         return colType;
     }
 
+    @Override
+    public MappedTable mapTable(final ResultSet tables, List<String> ignoreErrorList) throws SQLException {
+        final String tableName = tables.getString("TABLE_NAME");
+        if (ignoreErrorList.contains(POSTGRES_IGNORE_UPPERCASE_TABLES)) {
+            return super.mapTable(tables, ignoreErrorList);
+        }
+        if (containsUppercaseCharacter(tableName)) {
+            throw new IllegalArgumentException("Table " + tableName + " cannot be used in virtual schema. " +
+                    "Set property IGNORE_ERROR_LIST to POSTGRES_IGNORE_UPPERCASE_TABLES to enforce schema creation.");
+        } else {
+            return super.mapTable(tables, ignoreErrorList);
+        }
+    }
+
+    private boolean containsUppercaseCharacter(String tableName) {
+        return !tableName.equals(tableName.toLowerCase());
```

Comment:
This way to test for lower case is [locale-sensitive](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#toLowerCase--). If the user used a different locale than the default locale, this can lead to unexpected results.
That being said I am not sure other methods are able to circumvent that problem without knowing the original locale. Probably not.

Reply:
Will leave that to a special branch concerning utf8 and locale